### PR TITLE
Start page UI

### PR DESCRIPTION
**Pull request by fishyvishy**
Docstrings have not been added yet. This PR is to allow the team to test whether we can patch the startPage UI into the the user registration use case successfully.

### CODE REVIEW

**Review by EricKarpovits — APPROVED** [review]
Solid Work Vish, add the docstrings when you have time.

File: src/main/java/main.java [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+        JFrame application = new JFrame("RecipEZ");
+        application.setSize(550,200 );
+        application.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        CardLayout cardLayout = new CardLayout();
+        JPanel screens = new JPanel(cardLayout);
+        application.add(screens);
+
+        createUserController mockRegisterController = new createUserController();
 
+        startPageViewModel startViewModel = new startPageViewModel(mockRegisterController);
```
Comment (EricKarpovits) [inline src/main/java/main.java]: One comment  it should be startPageViewModelInterface

Comment (EricKarpovits) [reply src/main/java/main.java]: startPageViewModelInterface startViewModel = new startPageViewModel(mockRegisterController);

File: src/main/java/UI/startPageUI/registerUserView.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        this.viewModel = viewModel;
+        this.screens = screens;
+        this.screenLayout = screenLayout;
+        this.registerController = controller;
+
+        this.username = new JTextField(15);
+
+        this.register_response = new JTextField(30);
+        this.register_response.setEditable(false);
+        this.register_response.setOpaque(true);
```
Comment (Shaffaan) [inline src/main/java/UI/startPageUI/registerUserView.java:39]: Is the opaque always going to be true?

File: src/main/java/UI/startPageUI/registerUserView.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     * @param e the event to be processed
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getSource() == back) {
+            viewModel.pageState = "main";
+            screenLayout.show(screens, viewModel.pageState);
+        }
+        if (e.getSource() == create_user_btn) {
+            registerController.create(username.getText());;
```
Comment (Shaffaan) [inline src/main/java/UI/startPageUI/registerUserView.java]: good work here follows clean architecture principles

Comment (Shaffaan) [inline src/main/java/UI/startPageUI/registerUserView.java]: good work here follows clean architecture principles

File: src/main/java/UI/startPageUI/startPageViewMain.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        welcomePanel.add(title);
+
+        CardLayout screenLayout = new CardLayout();
+
+        JPanel startScreens = new JPanel();
+        startScreens.setLayout(screenLayout);
+
+        startPageViewWelcome welcome = new startPageViewWelcome(viewModel, registerController, startScreens, screenLayout);
+        registerUserView register = new registerUserView(viewModel, registerController, startScreens, screenLayout);
+        userLoginView login = new userLoginView(viewModel, registerController, startScreens, screenLayout);
```
Comment (Shaffaan) [inline src/main/java/UI/startPageUI/startPageViewMain.java:42]: We need to change this part, will create a new use case but for now its good.

**Review by Shaffaan — APPROVED** [review]
Really good work vish, will create another use case and once other pull reqs are reviewed and merged we can have our code up and running

**Review by rickychen175 — APPROVED** [review]
Great work. The starting pages look very nice.

File: src/main/java/UI/startPageUI/startPageViewModel.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+     */
+    public startPageViewModel() {
+        this.pageState = "main";
+    }
+
+    /**
+     * Updates view for successful user registration
+     * @param response successful user registration response
+     */
+    public void registration_success(String response) {
```
Comment (Zhuofan-Y) [inline src/main/java/UI/startPageUI/startPageViewModel.java:23]: Add an Override message when you are overriding methods from interface.

**Review by Zhuofan-Y — COMMENTED** [review]
Nice design! Nicely following design principles.

**Pull request merged**